### PR TITLE
Create PathTrackerProxy to get config_id from property directly

### PR DESCRIPTION
**Pull request by earafat-oak9**

### CODE REVIEW

File: py/core/types_tests.py [hunk 1]
```diff
@@ -81,5 +85,236 @@ def test_finding(self, finding: Finding, description: str):
...
+        listener_rules = [
+            ProxyTests.LoadBalancerListenerRule([
+                ProxyTests.LoadBalancer_ListenerRule_RuleCondition(
+                    ['myval1', 'myval2'],
+                    ProxyTests.LoadBalancer_ListenerRule_HostHeaderConfig(['X-CacheControl', 'X-CORS'])
+                )
+            ])
+        ]
+        actual_load_balancer = ProxyTests.LoadBalancer('my_lb', {'owner': 'joe'}, listeners, listener_rules)
+        self.validation_resource = PathTrackerProxy.create(actual_load_balancer)
```
Comment (earafat-oak9) [inline py/core/types_tests.py:146]: This is how the proxy object is used.
The proto object is passed to `PathTrackerProxy.create()` and the returned value is what we should perform the validation logic on.

File: py/core/types_tests.py [hunk 2]
```diff
@@ -81,5 +85,236 @@ def test_finding(self, finding: Finding, description: str):
...
+            self.assertEqual(prop_path, f'listeners[{idx}].target_group_id')
+            idx += 1
+
+    def test_prop_deeply_nested_string(self):
+        prop = self.validation_resource.listener_rules[0].conditions[0].host_header_config.values[0]
+
+        self.assertTrue(prop == 'X-CacheControl')
+        prop_path = prop._cur_path
+        print(prop_path)
+        self.assertEqual(prop_path, f'listener_rules[0].conditions[0].host_header_config.values[0]')
```
Comment (earafat-oak9) [inline py/core/types_tests.py:244]: Here's an example of how to get the config_id of a property.

`prop` is the property we're accessing so we can run validation logic on. And on that _seemingly string value_, you can do `prop._cur_path` to get the full path of that property as shown in the test.

Comment (earafat-oak9) [reply py/core/types_tests.py:244]: The idea is that we'll build a function to create design gaps, and all the user has to do is pass to this function the property on its own, and form that property we'll get the config_id and build the Finding object.

**Review by bnicoll-oak9 — APPROVED** [review]

**Review by jreiner-oak9 — APPROVED** [review]
_Ich verstehe nur Bahnhof._

**Review by morgan-oak9 — APPROVED** [review]

**Comment by earafat-oak9** [comment]
> _Ich verstehe nur Bahnhof._

Ich verstehe nur Englisch.

**Pull request merged**